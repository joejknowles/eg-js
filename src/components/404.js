import React from 'react';

export default () => (
    <h1>404 - page not found. Oops</h1>
);
