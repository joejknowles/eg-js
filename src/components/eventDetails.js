import React, { Component } from 'react';
import { connect } from 'react-redux';

import { createEventSelector } from '../reducers';
import { fetchEventDetailsRequest } from '../actions';

export class EventDetails extends Component {
  componentDidMount() {
    this.props.fetchEventDetails();
  }

  render() {
    const props = this.props;
    return (
     <div key={ props.id } className="event-details">
       <img src={ props.image } className="event-image" alt="event" />
       <h3 className="event-title">{ props.title }</h3>
       <p>
         { props.date }, at { props.venue }
       </p>
     </div>
   );
  }
}

const mapStateToProps = (state, { params: { id } }) => (
  createEventSelector(id)
);

const mapStateToDispatch = (dispatch, { params: { id } }) => ({
  fetchEventDetails: () => dispatch(fetchEventDetailsRequest(id))
});

export default connect(mapStateToProps, mapStateToDispatch)(EventDetails);
