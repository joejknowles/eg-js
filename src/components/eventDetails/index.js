import React, { Component } from 'react';
import { connect } from 'react-redux';

import Details from './details';

import { createEventSelector } from '../../reducers';
import { fetchEventDetailsRequest } from '../../actions';

export class EventDetails extends Component {
  componentDidMount() {
    this.props.fetchEventDetails();
  }

  render() {
    const props = this.props;
    return (
      <div key={ props.id } className="event-details">
        <img src={ props.image } className="event-image" alt="event" />
        <div className="event-info">
          <h2 className="event-title" dangerouslySetInnerHTML={ { __html: props.title } }/>
          <Details { ...props } />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, { params: { id } }) => (
  createEventSelector(id)
);

const mapStateToDispatch = (dispatch, { params: { id } }) => ({
  fetchEventDetails: () => dispatch(fetchEventDetailsRequest(id))
});

export default connect(mapStateToProps, mapStateToDispatch)(EventDetails);
