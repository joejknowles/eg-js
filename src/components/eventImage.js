import React from 'react';

const defaultImageUrl = 'https://placehold.it/100x100?text=IMAGE';

export default ({ url }) => (
  <img
    src={ url || defaultImageUrl }
    className="event-image"
    alt="event"
  />
);
