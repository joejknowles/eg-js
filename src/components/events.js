import React, { Component } from 'react';
import { connect } from 'react-redux';

import Listing from './listing';
import LoadingIndicator from './loadingIndicator';
import { eventsSelector, eventsLoadingSelector } from '../reducers'

export class Events extends Component {
  componentDidMount() {
    this.props.fetchEvents();
  }

  render() {
    const { props } = this;
    if (props.eventsLoading) return (<LoadingIndicator />);
    return (
      <div className="events-list">
        {
          props.events.map(
            id => <Listing id={ id } key={ id } />
          )
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  events: eventsSelector(state),
  eventsLoading: eventsLoadingSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchEvents: () => dispatch({ type: "FETCH_EVENTS_REQUEST" })
});

export default connect(mapStateToProps, mapDispatchToProps)(Events);
