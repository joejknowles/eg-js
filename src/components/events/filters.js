import React from 'react';

export const Filters = (props) => (
  <div className='Filters'>
    <input onChange={ event => props.onTypeFilterChange(event.target.value) } ></input>
  </div>
);
