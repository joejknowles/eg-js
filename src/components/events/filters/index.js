import React from 'react';
import { connect } from 'react-redux';

export const Filters = (props) => (
  <div className='Filters'>
    <select onChange={ event => props.onTypeFilterChange(event.target.value) } >
      <div></div>
    </select>
  </div>
);

const mapStateToDispatch = (dispatch) => ({
  onTypeFilterChange: (filter) => dispatch({ type: 'SET_TYPE_FILTER', filter})
});

export default connect(undefined, mapStateToDispatch)(Filters);
