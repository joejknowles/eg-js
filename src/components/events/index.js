import React, { Component } from 'react';
import { connect } from 'react-redux';

import Listing from '../listing';
import LoadingIndicator from '../loadingIndicator';
import { eventsFirstLoadingSelector } from '../../reducers';
import { filteredEventsSelector } from '../../selectors/filter'

export class Events extends Component {
  componentDidMount() {
    this.props.fetchEvents();
  }

  render() {
    const { props } = this;
    if (props.eventsLoading) return (<LoadingIndicator />);
    return (
      <div className="events-list">
        {
          props.events.map(
            id => <Listing id={ id } key={ id } />
          )
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  events: filteredEventsSelector(state),
  eventsLoading: eventsFirstLoadingSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchEvents: () => dispatch({ type: "FETCH_EVENTS_REQUEST" })
});

export default connect(mapStateToProps, mapDispatchToProps)(Events);
