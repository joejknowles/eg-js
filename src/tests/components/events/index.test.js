import { Events } from '../../../components/events';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import { shallow } from 'enzyme';

const shallowRenderer = ReactTestUtils.createRenderer();

describe('Events Component', () => {
  it('renders as before with an event', () => {
    const component = shallowRenderer.render(<Events events={ [ 1 ] } />);
    expect(component).toMatchSnapshot();
  });

  it('renders as before with no events', () => {
    const component = shallowRenderer.render(<Events events={ [] } />);
    expect(component).toMatchSnapshot();
  });

  it('renders as before while loading', () => {
    const component = shallowRenderer.render(<Events eventsLoading={ true } />);
    expect(component).toMatchSnapshot();
  });

  it('calls fetchEvents on componentDidMount', () => {
    const fetchEventsMock = jest.fn();
    const component = shallow(
      <Events
        events={ [] }
        fetchEvents={ fetchEventsMock }
      />
    );
    component.instance().componentDidMount();
    expect(fetchEventsMock).toHaveBeenCalled();
  });
});
