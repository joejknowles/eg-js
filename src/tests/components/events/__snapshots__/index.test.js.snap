exports[`Events Component renders as before 1`] = `
<div
  className="events-list">
  
</div>
`;
